azure-devops/azext_devops/dev/repos/policy.py

```diff
@@ -0,0 +1,371 @@
+    :param displayName : Display Name. Required if policy type is Buildpolicy.
+    :type displayName : string
+    :param validDuration : Valid duration. Required if policy type is Buildpolicy.
+    :type validDuration : double
+
+    :param useSquashMerge: Use Squash Merge. Required if policy type is MergeStrategyPolicy
+    :type useSquashMerge: bool
+
+    :param maximumGitBlobSizeInBytes: Maximum Git Blob Size In Bytes. Required if policy type is FileSizePolicy
+    :type maximumGitBlobSizeInBytes: long
+    :param useUncompressedSize: Use uncompressed size. Required if policy type is FileSizePolicy
+    :type useUncompressedSize: bool
+
+    :param organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
+    :type organization: str
+    :param project: Name or ID of the project.
+    :type project: str
+    :param detect: Automatically detect organization. Default is "on".
+    :type detect: str
+    :rtype: :class:`<PolicyConfiguration> <policy.v4_0.models.PolicyConfiguration>`
+    """
+    try:
+        organization, project = resolve_instance_and_project(
+            detect=detect, organization=organization, project=project)
+        policy_client = get_policy_client(organization)
+
+        policyConfigurationToCreate = generateConfigurationObject(
+            repository_id, branch,
+            policy_type,
+            isBlocking, isEnabled,
+            minimumApproverCount, creatorVoteCounts, allowDownvotes, resetOnSourcePush,
+            useSquashMerge,
+            buildDefinitionId, queueOnSourceUpdateOnly, manualQueueOnly, displayName, validDuration,
+            maximumGitBlobSizeInBytes, useUncompressedSize,
+            optionalReviewerIds, requiredReviewerIds, message,
+            organization)
+
+        return policy_client.create_policy_configuration(configuration=policyConfigurationToCreate, project=project)
+    except VstsServiceError as ex:
+        raise CLIError(ex)
+
+# pylint: disable=too-many-locals
+def update_policy(repository_id, branch,
+                  policy_id,
+                  isBlocking=False, isEnabled=False,
+                  policy_type=None,
+                  minimumApproverCount=None, creatorVoteCounts=None, allowDownvotes=None, resetOnSourcePush=None,
+                  useSquashMerge=None,
+                  buildDefinitionId=None, queueOnSourceUpdateOnly=None, manualQueueOnly=None, displayName=None, validDuration=None,
+                  maximumGitBlobSizeInBytes=None, useUncompressedSize=None,
+                  optionalReviewerIds=None, requiredReviewerIds=None, message=None,
+                  organization=None, project=None, detect=None):
+    """
+    :param repository_id: Id (UUID) of the repository on which to apply the policy to.
+    :type repository_id: string
+    :param branch: Branch on which this policy should be applied
+    :type branch: string
+    :param policy_id: Policy Id of policy which needs to be updated
```

Comment:
ID of the policy which needs to be updated

Reply:
done